### PR TITLE
Develop rebased on 3e4e7be

### PR DESCRIPTION
**Pull request by g11tech**
rebased on
```
commit 3e4e7bed681bfa6d1c95abbcc28f6508e828339d (HEAD -> master, origin/master, origin/HEAD)
Author: Holger Drewes <Holger.Drewes@gmail.com>
Date:   Thu Jun 2 22:14:19 2022 +0200

    Blockchain: Clique-recently-signed False Positive Fix (#1931)
    

```

### CODE REVIEW

**Comment by holgerd77** [comment]
Oh, that's so so great, thanks a lot Gajinder!!! ❤️ 🙏 🚀

To prepare for the subsequent process: so there is this *last* `master` commit from the v5 releases PR still missing, otherwise this will be the very last time that we have rebased anything here (so: at least in this `develop` towards `master` directly, might be some `develop-feature-branch` -> `develop` rebases still be necessary).

I guess (hope 😬) this will be easily integratable here (basically by an additional rebase towards this last commit, should not be to "conflicty" since only/mostly doc files and package.json files are touched?).

So: I will basically merge "right now" (or in the next 30 minutes or so).

Side note: I guess we might be early here then - depending on how the rebase is going, will likely be a tricky last one - and eventually directly switch onto the newly rebased `develop` branch from here tomorrow (so being *early* according to our planning issue actually 😋: #1914)

**Comment by holgerd77** [comment]
Update: merged in the release PR (so: #1927)

**Comment by holgerd77** [comment]
Another update: if you re-rebase 😋: wait a minute, I guess I would have needed to submit an updated `package-lock.json` at the end of the version number updates of the release PR. Will do this in a small separate PR (which I would also admin-merge). Will give an update here.

**Comment by holgerd77** [comment]
I've got a bug in the client which blocks me from doing the release round: https://github.com/ethereumjs/ethereumjs-monorepo/issues/1930

If you have some capacity right now (or relatively soon) and could re-prioritize and have a look that would be appreciated! 🙂

**Comment by holgerd77** [comment]
(can be fixed just by a normal PR towards master)

**Comment by holgerd77** [comment]
I now merged in #1924. This would need to be cherry-picked on top here (git cherry-pick d62b7f47c04e4697afd081be215cc2b468366a95) once the PR is generally ready.

(@g11tech let me know if these kind of merges pose any problems and you would prefer an alternative procedure to go forward here right now)

**Comment by g11tech** [comment]
> I now merged in #1924. This would need to be cherry-picked on top here (git cherry-pick [d62b7f4](https://github.com/ethereumjs/ethereumjs-monorepo/commit/d62b7f47c04e4697afd081be215cc2b468366a95)) once the PR is generally ready.
> 
> (@g11tech let me know if these kind of merges pose any problems and you would prefer an alternative procedure to go forward here right now)

cherry picking sounds good @holgerd77  :+1:

**Comment by holgerd77** [comment]
I've now also merged in #1867, would also need to be cherry-picked (`git cherry-pick a7b1648bc40ee6bd2f33732017530d27c14be27e`).

This is the *last* PR I would merge, so we are now on both the final (for the rebase) `develop` AND `master` state. So once this is working we can take this over as the new (and: last) `develop` state! 💯 🎉

Will now do the last round of v5 releases and also branch off the `v5-maintenance` branch.

**Comment by holgerd77** [comment]
Oh, that already looks pretty good! 🙂

File: packages/tx/src/legacyTransaction.ts [hunk 1]
```diff
@@ -356,22 +356,13 @@ export default class Transaction extends BaseTransaction<Transaction> {
...
-      // v is 1. not matching the EIP-155 chainId included case and...
-      // v is 2. not matching the classic v=27 or v=28 case
-      if (v.ltn(37) && !v.eqn(27) && !v.eqn(28)) {
-        throw new Error(
-          `Legacy txs need either v = 27/28 or v >= 37 (EIP-155 replay protection), got v = ${v}`
-        )
-      }
-    }
-
-    let chainIdBN
```
Comment (holgerd77) [inline packages/tx/src/legacyTransaction.ts:392]: This has been fixed in the "wrong direction", so the removed code is actually the correct one, see #1905 for context and discussion.

So fix should be just to replace `v.ltn(37) && !v.eqn(27) && !v.eqn(28)` with the normal BigInt-operator comparisons and literally nothing else.

(let me know if I overlook something)

Comment (holgerd77) [reply packages/tx/src/legacyTransaction.ts:392]: Update: ah, just seeing that Andrew addressed this in a follow-up commit, so not an issue.

File: packages/blockchain/src/index.ts [hunk 2]
```diff
@@ -1149,6 +1151,30 @@ export default class Blockchain implements BlockchainInterface {
...
+        break
+      case ConsensusAlgorithm.Ethash:
+        if (!(this.consensus instanceof EthashConsensus)) {
+          this.consensus = new EthashConsensus({ blockchain: this })
+        }
+        break
+      default:
+        throw new Error(`consensus algorithm ${this._common.consensusAlgorithm()} not supported`)
+    }
+  }
```
Comment (holgerd77) [inline packages/blockchain/src/index.ts:1176]: Whew, that's a hard one: a whole new API method introduced in a rebase-fix-PR. 😋 😬

Are we confident enough that we want to expose? Or would we rather want to keep this private (or rather: protected) for now?

Comment (g11tech) [reply packages/blockchain/src/index.ts:1176]: yea it was tricky one to find! well will move to `protected` :+1:

**Comment by holgerd77** [comment]
Branch has been applied to master via #1943, will close.

**Pull request closed without merging**